src/contract.rs

```diff
@@ -527,30 +612,54 @@ fn get_bank_transfer_to_msg(recipient: &Addr, denom: &str, native_amount: Uint12
     transfer_bank_cosmos_msg
 }
 
+fn fee_decimal_to_uint128(decimal: Decimal) -> StdResult<Uint128> {
+    let result: Uint128 = decimal
+        .atomics()
+        .checked_mul(FEE_SCALE_FACTOR)
+        .map_err(StdError::overflow)?;
+
+    Ok(result / FEE_DECIMAL_PRECISION)
+}
+
 fn get_input_price(
     input_amount: Uint128,
     input_reserve: Uint128,
     output_reserve: Uint128,
+    fee_percent: Decimal,
 ) -> StdResult<Uint128> {
     if input_reserve == Uint128::zero() || output_reserve == Uint128::zero() {
         return Err(StdError::generic_err("No liquidity"));
     };
 
-    let input_amount_with_fee = input_amount
-        .checked_mul(Uint128::new(997))
-        .map_err(StdError::overflow)?;
+    let fee_percent = fee_decimal_to_uint128(fee_percent)?;
```

Comment:
it might be worth adding a comment about fee percent versus fraction with percent being 0-100 vs fee fraction being 0-1